src/main/java/ru/yandex/clickhouse/response/ClickHouseResultSet.java

```diff
@@ -592,32 +646,113 @@ public BigDecimal getBigDecimal(int columnIndex, int scale)  {
 
     @Override
     public void setFetchDirection(int direction) throws SQLException {
-        // ignore perfomance hint
+        // ignore performance hint
     }
 
     @Override
     public void setFetchSize(int rows) throws SQLException {
-        // ignore perfomance hint
+        // ignore performance hint
+    }
+
+    static Map<String, String> getTableNames(String query, String[] columnNames) {
+        Map<String, String> tableNames = new HashMap<String, String>(columnNames.length);
```

Comment:
There also must be a more robust query parser handling comments and quoted strings.
I guess there must be one for the driver that parses whole query to tokens(maybe even generated with some parser generator).

Reply:
As I see it is a separate feature from isNullable, so it is better to extract it to a separate PR if possible.